docs/content/en/docs/Reference/cloud_init.md

```diff
@@ -85,7 +85,86 @@ stages:
 The default cloud-config format is split into *stages* (*initramfs*, *boot*, *network*, *initramfs*, *reconcile*, called generically **STAGE_ID** below) [see also stages](../../customizing/stages) that are emitted internally during the various phases by calling `cos-setup STAGE_ID`. 
 *steps* (**STEP_NAME** below) defined for each stage are executed in order.
 
-Each cloud-config file is loaded and executed only at the apprioriate stage, this allows further components to emit their own stages at the desired time.
+Each cloud-config file is loaded and executed only at the appropriate stage, this allows further components to emit their own stages at the desired time.
+
+_Note_:
+- The execution order of multiple `substeps` within a single `step` is not guaranteed, especially whether the `commands` are executed after other substeps. For example:
+  ```
+  stages:
+    fs:
+      - name: test
+        environment_file: /tmp/env_file1
+        environment:
+          ENV1: this is ENV1
+        files:
+          - path: /tmp/file1
+            content: "this is file1\n"
+        directories:
+          - path: /tmp/dir1
+        users:
+          user1:
+              passwd: password
+        commands:
+          - |
+            bash -x >> /tmp/log 2>&1 <<'EOF'
+            ls -lFd /tmp/env_file1
+            printenv ENV1
+            ls -lFd /tmp/file1
+            ls -lFd /tmp/dir1
+            id user1
+            EOF
+  ```
+  
+  You might expect that when the `commands` get executed, all things defined above it has been executed, such as the user1 has been created.
```

Comment:
As described before the execution ordering is specified when loading the plugins in https://github.com/rancher/elemental-toolkit/blob/d8450e01b9119b76670421880070d2cc7fa5f936/pkg/cloudinit/cloudinit.go#L52

Reply:
@frelon  Added at https://github.com/rancher/elemental-toolkit/pull/2294/files#r2256178003